### PR TITLE
Move some `build-pass` tests to `check-pass`

### PR DESCRIPTION
**Pull request by LeSeulArtichaut**
Helps with #62277.

r? @cramertj cc @Centril

### CODE REVIEW

**Comment by Centril** [comment]
cc @rust-lang/wg-const-eval  since there are a lot of const related tests here

**Comment by JohnTitor** [comment]
Let's re-assigning someone from wg-const-eval...
r? @RalfJung maybe?

**Comment by RalfJung** [comment]
@wesleywiser @oli-obk does const_prop run in check-pass?

**Comment by RalfJung** [comment]
As far as I know, all things const-eval happen even for check builds, so everything in `src/test/ui/consts` is likely fine (I asked for confirmation above).

But for ` src/test/rustdoc-ui` and the other tests I'm afraid I have no idea. @rust-lang/compiler we need someone with wide-spread compiler knowledge I think.

**Comment by Centril** [comment]
`check-pass` means `--emit metadata` if that helps.

**Comment by Centril** [comment]
Maybe we should split the PR into e.g. const parts, rustdoc-ui parts, and then the various experts can review.

**Comment by wesleywiser** [comment]
> @wesleywiser @oli-obk does const_prop run in check-pass?

Yes, it should.

**Comment by LeSeulArtichaut** [comment]
@RalfJung @Centril would it be more convenient to break the PR into multiple parts? If so, should I keep this PR open for, say, all const-eval related tests, and open a new PR with the other ones? Or should I close this and open 2 new PR's?

**Comment by RalfJung** [comment]
Leaving one part of the tests in this PR is fine. Feel free to force-push for that.

**Comment by LeSeulArtichaut** [comment]
Done!
*Hehe the PR went from 72 changed files to 20*
Poke @RalfJung

File: src/test/ui/consts/const-eval/issue-51300.rs [hunk 1]
```diff
@@ -1,4 +1,4 @@
-// build-pass (FIXME(62277): could be check-pass?)
+// build-pass
```
Comment (RalfJung) [inline src/test/ui/consts/const-eval/issue-51300.rs]: Why did you make this build-pass?

Comment (RalfJung) [reply src/test/ui/consts/const-eval/issue-51300.rs]: (And there are some more.)

Comment (LeSeulArtichaut) [reply src/test/ui/consts/const-eval/issue-51300.rs]: I re-read all of the `build-pass` ones, and I took the time to do more research about the issues they prevent regression to. All in all I changed the all to `check-pass`

**Comment by RalfJung** [comment]
Thanks!

@bors r+

**Pull request merged**